In [4]:
%matplotlib inline

In [6]:
# Imports
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# Data Exploration
## Of Asthma Disease Dataset

In this notebook, we will read the provided dataset (from [here](https://www.kaggle.com/datasets/rabieelkharoua/asthma-disease-dataset)) and perform EDA on it. 

In [11]:
asthma_data = pd.read_csv("https://raw.githubusercontent.com/ShopyEsladuk/Asthma-Diagnosis/main/data/asthma_disease_data.csv")

In [13]:
asthma_data

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
0          5034   63       0          1               0  15.848744        0   
1          5035   26       1          2               2  22.757042        0   
2          5036   57       0          2               1  18.395396        0   
3          5037   40       1          2               1  38.515278        0   
4          5038   61       0          0               3  19.283802        0   
...         ...  ...     ...        ...             ...        ...      ...   
2387       7421   43       1          0               2  29.059613        0   
2388       7422   18       1          0               1  20.740850        0   
2389       7423   54       0          3               2  37.079560        0   
2390       7424   46       1          0               2  23.444712        0   
2391       7425   26       1          0               0  28.123021        1   

      PhysicalActivity  DietQuality  SleepQuality  ...  LungFunctionFEV1  \
0             0.894448     5.488696      8.701003  ...          1.369051   
1             5.897329     6.341014      5.153966  ...          2.197767   
2             6.739367     9.196237      6.840647  ...          1.698011   
3             1.404503     5.826532      4.253036  ...          3.032037   
4             4.604493     3.127048      9.625799  ...          3.470589   
...                ...          ...           ...  ...               ...   
2387          3.019854     6.119637      8.300960  ...          3.125249   
2388          5.805180     4.386992      7.731192  ...          1.132977   
2389          4.735169     8.214064      7.483521  ...          1.685962   
2390          9.672637     7.362861      6.717272  ...          3.481549   
2391          1.613138     7.412878      8.512253  ...          2.280613   

      LungFunctionFVC  Wheezing  ShortnessOfBreath  ChestTightness  Coughing  \
0            4.941206         0                  0               1         0   
1            1.702393         1                  0               0         1   
2            5.022553         1                  1               1         0   
3            2.300159         1                  0               1         1   
4            3.067944         1                  1               1         0   
...               ...       ...                ...             ...       ...   
2387         5.166032         0                  1               0         0   
2388         5.509502         0                  0               0         1   
2389         3.346877         1                  0               1         1   
2390         1.713274         0                  1               1         0   
2391         2.453284         1                  1               0         0   

      NighttimeSymptoms  ExerciseInduced  Diagnosis  DoctorInCharge  
0                     0                1          0       Dr_Confid  
1                     1                1          0       Dr_Confid  
2                     1                1          0       Dr_Confid  
3                     1                0          0       Dr_Confid  
4                     0                1          0       Dr_Confid  
...                 ...              ...        ...             ...  
2387                  0                1          1       Dr_Confid  
2388                  1                0          1       Dr_Confid  
2389                  0                1          1       Dr_Confid  
2390                  1                1          0       Dr_Confid  
2391                  1                0          0       Dr_Confid  

[2392 rows x 29 columns]

In [37]:
asthma_data.dtypes

PatientID                   int64
Age                         int64
Gender                      int64
Ethnicity                   int64
EducationLevel              int64
BMI                       float64
Smoking                     int64
PhysicalActivity          float64
DietQuality               float64
SleepQuality              float64
PollutionExposure         float64
PollenExposure            float64
DustExposure              float64
PetAllergy                  int64
FamilyHistoryAsthma         int64
HistoryOfAllergies          int64
Eczema                      int64
HayFever                    int64
GastroesophagealReflux      int64
LungFunctionFEV1          float64
LungFunctionFVC           float64
Wheezing                    int64
ShortnessOfBreath           int64
ChestTightness              int64
Coughing                    int64
NighttimeSymptoms           int64
ExerciseInduced             int64
Diagnosis                   int64
DoctorInCharge             object
dtype: object

In [41]:
asthma_data.DoctorInCharge.describe()

count          2392
unique            1
top       Dr_Confid
freq           2392
Name: DoctorInCharge, dtype: object

In [35]:
asthma_data.describe().T

count         mean         std          min  \
PatientID               2392.0  6229.500000  690.655244  5034.000000   
Age                     2392.0    42.137960   21.606655     5.000000   
Gender                  2392.0     0.493311    0.500060     0.000000   
Ethnicity               2392.0     0.669732    0.986120     0.000000   
EducationLevel          2392.0     1.307274    0.898242     0.000000   
BMI                     2392.0    27.244877    7.201628    15.031803   
Smoking                 2392.0     0.141722    0.348838     0.000000   
PhysicalActivity        2392.0     5.051786    2.903574     0.001740   
DietQuality             2392.0     5.022867    2.909980     0.003031   
SleepQuality            2392.0     7.019012    1.732475     4.001437   
PollutionExposure       2392.0     5.013030    2.935605     0.001022   
PollenExposure          2392.0     5.114221    2.893088     0.000659   
DustExposure            2392.0     4.981864    2.859192     0.002434   
PetAllergy              2392.0     0.165970    0.372131     0.000000   
FamilyHistoryAsthma     2392.0     0.301003    0.458790     0.000000   
HistoryOfAllergies      2392.0     0.399247    0.489846     0.000000   
Eczema                  2392.0     0.191890    0.393869     0.000000   
HayFever                2392.0     0.253344    0.435017     0.000000   
GastroesophagealReflux  2392.0     0.158027    0.364842     0.000000   
LungFunctionFEV1        2392.0     2.548564    0.861809     1.000459   
LungFunctionFVC         2392.0     3.741270    1.303689     1.500045   
Wheezing                2392.0     0.596154    0.490770     0.000000   
ShortnessOfBreath       2392.0     0.500418    0.500104     0.000000   
ChestTightness          2392.0     0.503344    0.500093     0.000000   
Coughing                2392.0     0.503344    0.500093     0.000000   
NighttimeSymptoms       2392.0     0.602425    0.489499     0.000000   
ExerciseInduced         2392.0     0.604933    0.488967     0.000000   
Diagnosis               2392.0     0.051839    0.221749     0.000000   

                                25%          50%          75%          max  
PatientID               5631.750000  6229.500000  6827.250000  7425.000000  
Age                       23.000000    42.000000    61.000000    79.000000  
Gender                     0.000000     0.000000     1.000000     1.000000  
Ethnicity                  0.000000     0.000000     1.000000     3.000000  
EducationLevel             1.000000     1.000000     2.000000     3.000000  
BMI                       20.968313    27.052202    33.555903    39.985611  
Smoking                    0.000000     0.000000     0.000000     1.000000  
PhysicalActivity           2.578333     5.016881     7.540234     9.995809  
DietQuality                2.432043     5.115383     7.544216     9.999904  
SleepQuality               5.498500     6.975839     8.526950     9.996235  
PollutionExposure          2.432559     5.036519     7.630037     9.998964  
PollenExposure             2.599996     5.098452     7.708836     9.999555  
DustExposure               2.467489     4.903278     7.432218     9.999708  
PetAllergy                 0.000000     0.000000     0.000000     1.000000  
FamilyHistoryAsthma        0.000000     0.000000     1.000000     1.000000  
HistoryOfAllergies         0.000000     0.000000     1.000000     1.000000  
Eczema                     0.000000     0.000000     0.000000     1.000000  
HayFever                   0.000000     0.000000     1.000000     1.000000  
GastroesophagealReflux     0.000000     0.000000     0.000000     1.000000  
LungFunctionFEV1           1.824113     2.553244     3.292897     3.999719  
LungFunctionFVC            2.607489     3.734982     4.864121     5.999421  
Wheezing                   0.000000     1.000000     1.000000     1.000000  
ShortnessOfBreath          0.000000     1.000000     1.000000     1.000000  
ChestTightness             0.000000     1.000000     1.000000     1.000000  
Coughing   

Seems to have no missing data. Also, the shape is (2392, 29). It is clear it needs some tidying and cleaning.